### ISSUE TITLE
false-positive: button-name fails on img with role=button

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#2577]
This was brought to our attention. The following code fails the `button-name` rule as that rule does not look at `alt` attributes. All our supported AT/browsers read the `alt` of the image correctly.

```html
<img src="foo.png" tabindex="0" alt="my custom img" role="button"/>
```

**Comment by padmavemulapati** [comment]
Verified with the fix in latest axe-core(4.0.2v) dated 2.10.2020
Not seeing `button-name` rule failure. At the same time was able to reproduce it with 3.5.4
![image](https://user-images.githubusercontent.com/47409579/97836048-7335a380-1d01-11eb-8170-b5a2e1c10e05.png)
